Sources/OutlineView/TreeMap.swift

```diff
@@ -0,0 +1,223 @@
+
+/// An object representing the state of an OutlineView in which the content
+/// of the OutlineView is Identifiable. The TreeMap should be able to stay
+/// in sync with the OutlineView by reacting to data updates in the OutlineView.
+class TreeMap<D: Hashable> {
+        
+    struct Node {
+        var parentId: D?
+        var isLeaf: Bool
+        
+        // childIds can be nil if `isLeaf` == true, or if the item is not expanded.
+        // If childIds is not nil, that means the item is expanded.
+        var childIds: [D]?
+    }
+    
+    private (set) var rootData: [D] = []
+    private var directory: [D : Node] = [:]
+        
+    init() {}
+    
+    @available(macOS 10.15, *)
```

Comment:
You can move this availability annotation to the TreeMap type declaration.

Reply:
Fixed that here: https://github.com/Sameesunkaria/OutlineView/pull/10/commits/7b8e914a9d9ccdec3a9fd55823cc1613bbc44d46